### PR TITLE
Implemented Browser tests

### LINKED ISSUE #185: Implement Browser tests
https://github.com/GoogleChrome/puppeteer/blob/v1.0.0/test/test.js#L268

### PR DESCRIPTION
**Pull request by Meir017**
closes #185

### CODE REVIEW

File: lib/PuppeteerSharp.Tests/BrowserTests/Events/DisconnectedTests.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            var disconnectedRemote2 = 0;
+            originalBrowser.Disconnected += (sender, e) => ++disconnectedOriginal;
+            remoteBrowser1.Disconnected += (sender, e) => ++disconnectedRemote1;
+            remoteBrowser2.Disconnected += (sender, e) => ++disconnectedRemote2;
+
+            remoteBrowser2.Disconnect();
+            await Task.Delay(150);
+            Assert.Equal(0, disconnectedOriginal);
+            Assert.Equal(0, disconnectedRemote1);
+            Assert.Equal(1, disconnectedRemote2);
```
Comment (Meir017) [inline lib/PuppeteerSharp.Tests/BrowserTests/Events/DisconnectedTests.cs]: something isn't working with the `Disconnected` event

File: lib/PuppeteerSharp/Browser.cs [hunk 2]
```diff
@@ -97,6 +100,12 @@ internal void ChangeTarget(Target target)
             return version.product.ToString();
         }
 
+        public async Task<string> GetUserAgentAsync()
```
Comment (kblok) [inline lib/PuppeteerSharp/Browser.cs]: Docs

File: lib/PuppeteerSharp/Browser.cs [hunk 3]
```diff
@@ -116,7 +125,6 @@ internal void ChangeTarget(Target target)
             }
 
             Disconnect();
-            Closed?.Invoke(this, new EventArgs());
```
Comment (kblok) [inline lib/PuppeteerSharp/Browser.cs]: Could you explain this?

File: lib/PuppeteerSharp/Session.cs [hunk 4]
```diff
@@ -152,7 +148,7 @@ internal void OnMessage(string message)
             }
         }
 
-        internal void Close()
+        internal void OnClose()
```
Comment (kblok) [inline lib/PuppeteerSharp/Session.cs]: I don't know if I buy this rename 🤔

**Comment by kblok** [comment]
Why are we including `lib/Win64-526987/chrome-win32/locales/ar.pak`?

File: lib/PuppeteerSharp/Session.cs [hunk 5]
```diff
@@ -152,7 +148,7 @@ internal void OnMessage(string message)
             }
         }
 
-        internal void Close()
+        internal void OnClosed()
```
Comment (Meir017) [inline lib/PuppeteerSharp/Session.cs]: renamed to match - https://github.com/GoogleChrome/puppeteer/blob/v1.0.0/lib/Connection.js#L212

Comment (kblok) [reply lib/PuppeteerSharp/Session.cs]: Awesome

**Comment by kblok** [comment]
Check this out @Meir017  https://github.com/kblok/puppeteer-sharp/commit/f7ed88ea1d0cad13f60f651995ffa882f16dfa72

**Comment by kblok** [comment]
@Meir017 https://github.com/kblok/puppeteer-sharp/pull/224/commits/90e9b23f8fe498635a05b92525c30f1171c67e91

If you want to read a little bit about the WebSocket mess go to https://github.com/dotnet/corefx/pull/26429 and https://github.com/dotnet/corefx/issues/26556

I found out that we don't need to filter exceptions if the read fails we need to close and move on.

**Comment by kblok** [comment]
It's not WIP anymore right @Meir017 ?

**Pull request merged**